src/wasm-ast-writer.c

```diff
@@ -335,6 +332,9 @@ static void write_const(Context* ctx, const WasmConst* const_) {
       char buffer[128];
       wasm_write_float_hex(buffer, 128, const_->f32_bits);
       write_puts_space(ctx, buffer);
+      float f32;
```

Comment:
Type-punning always makes me a bit nervous, even though it's probably OK. Anyway, taking a look at the disassembly, gcc (at -O1) and clang (at -O0) optimize away the memcpy here already.